### PR TITLE
feat(query): update query server config

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- add max_memory_limit_enabled query config
- rename max_memory_usage to max_server_memory_usage

Closes #issue

### CODE REVIEW

**Comment by BohuTANG** [comment]
unit test need fixed :)

File: src/binaries/query/main.rs [hunk 1]
```diff
@@ -63,9 +63,11 @@ async fn main_entrypoint(mem_tracker: Arc<MemoryTracker>) -> Result<()> {
     init_default_metrics_recorder();
     set_panic_hook();
 
-    let size = conf.query.max_memory_usage as i64;
-    info!("Set memory limit: {}", size);
-    mem_tracker.set_limit(size);
+    if conf.query.max_memory_limit_enabled {
+        let size = conf.query.max_memory_usage as i64;
```
Comment (sundy-li) [inline src/binaries/query/main.rs:67]: why not use `max_memory_usage = 0` instead.

Comment (zhang2014) [reply src/binaries/query/main.rs:67]: The max_memory_usage is referenced otherwhere.

File: src/query/service/tests/it/storages/testdata/system-tables.txt [hunk 2]
```diff
@@ -348,6 +349,7 @@ DB.Table: 'system'.'settings', Table: settings-table_id:1, ver:0, Engine: System
 | load_file_metadata_expire_hours | 168        | 168         | SESSION | How many hours will the COPY file metadata expired in the metasrv, default value: 24*7=7days                                                                           | UInt64 |
 | max_block_size                  | 65536      | 65536       | SESSION | Maximum block size for reading, default value: 65536.                                                                                                                  | UInt64 |
 | max_execute_time                | 0          | 0           | SESSION | The maximum query execution time. it means no limit if the value is zero. default value: 0.                                                                            | UInt64 |
+| max_memory_usage                | 1073741824 | 27487790694 | SESSION | The maximum memory usage for processing single query, in bytes. By default the value is determined automatically.                                                      | UInt64 |
```
Comment (BohuTANG) [inline src/query/service/tests/it/storages/testdata/system-tables.txt:352]: Here should be `max_server_memory_usage`.

Comment (zhang2014) [reply src/query/service/tests/it/storages/testdata/system-tables.txt:352]: This is settings table.

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
This PR rename `max_memory_usage` to `max_server_memory_usage` cc @ZhiHanZ @everpcpc

**Pull request merged**